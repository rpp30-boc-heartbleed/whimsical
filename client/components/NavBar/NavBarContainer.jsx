import React from 'react';
import {
  View,
  Text,
  StyleSheet,
  TextInput,
  StatusBar,
  Button,
  TouchableOpacity,
} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';

const NavBarContainer = ({ navigation }) => {
  return (
    <View style={styles.container}>
      {/* <Text>This is the NavBar Component</Text> */}
      {/* Profile */}
      <TouchableOpacity>
        <Icon size={30} name='user' onPress={() => navigation.push('UserProfile')} />
      </TouchableOpacity>
      {/* Add Errand */}
      <TouchableOpacity>
        <Icon size={30} name='plus-circle' onPress={() => navigation.push('NewErrand')} />
      </TouchableOpacity>
      {/* FriendsList */}
      <TouchableOpacity>
        <Icon size={30} name='users' onPress={() => navigation.push('FriendsList')} />
      </TouchableOpacity>
      {/* Map */}
      <TouchableOpacity>
        <Icon size={30} name='map-marker' onPress={() => navigation.push('ErrandList')} />
      </TouchableOpacity>
      {/* Dashboard */}
      <TouchableOpacity>
        <Icon size={30} name='home' onPress={() => navigation.navigate('Dashboard')} />
      </TouchableOpacity>
      {/* <Button title="Go to Dashboard" onPress={() => navigation.navigate('Dashboard')} /> */}
      {/* <Button title="Go back" onPress={() => navigation.goBack()} /> */}
      <StatusBar />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#0782F9',
    justifyContent: 'space-between',
    flexDirection: 'row',
    padding: 25,
    marginTop: 50,
  },
});

export default NavBarContainer;
