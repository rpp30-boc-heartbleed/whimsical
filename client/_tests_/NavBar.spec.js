import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react-native';
import NavBarContainer from '../components/NavBar/NavBarContainer';

jest.mock('react-native-vector-icons/FontAwesome', () => 'Icon');

describe('NavBar', () => {
  // it('should render properly', () => {
  //   // snapshot testing
  //   // eslint-disable-next-line react/jsx-filename-extension
  //   const tree = renderer.create(<NavBarContainer />).toJSON();
  //   expect(tree).toMatchSnapshot();
  // });

  it('should render properly with icons', () => {
    const tree = renderer.create(<NavBarContainer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
