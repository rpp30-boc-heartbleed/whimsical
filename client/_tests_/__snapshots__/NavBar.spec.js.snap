// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should render properly with icons 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#0782F9",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginTop": 50,
      "padding": 25,
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Icon
      name="user"
      onPress={[Function]}
      size={30}
    />
  </View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Icon
      name="plus-circle"
      onPress={[Function]}
      size={30}
    />
  </View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Icon
      name="users"
      onPress={[Function]}
      size={30}
      testIID="users"
    />
  </View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Icon
      name="shopping-basket"
      onPress={[Function]}
      size={30}
    />
  </View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Icon
      name="home"
      onPress={[Function]}
      size={30}
    />
  </View>
</View>
`;
